src/v/base/fmt.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+#include <fmt/core.h>
+#include <fmt/ostream.h>
+
+#if FMT_VERSION >= 90100
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+#define RP_OSTREAM_FMT(type)                                                   \
+    template<>                                                                 \
+    struct fmt::formatter<type> : fmt::ostream_formatter {};
+
+#define RP_FMT_STREAMED(x) fmt::streamed(x)
```

Comment:
Some slight docs for these, even if they are just a copy/paste of the fmt docs might be useful so one doesn't have to look them up...